modules/ting_das/ting_das.module

```diff
@@ -0,0 +1,378 @@
+      ),
+      '#weight' => 15,
+    );
+  }
+  else {
+    $form['title'] = array(
+      '#markup' => '<div class="popupbar-title"><strong>' . t('Not resident in the municipality') . '</strong></div>',
+    );
+
+    $form['description'] = array(
+      '#prefix' => '<div class="form-item form-type-markup description">',
+      '#markup' => t('Only resident in the municipality is allowed to order digital copies.'),
+      '#suffix' => '</div>',
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Mail validation.
+ */
+function ting_das_mail_validate($element, $form_state) {
+  $email = $element['#value'];
+  if (!empty($email)) {
+    if (!ding_base_validate_email($email)) {
+      form_set_error($element['#name'], t('Please fill in a valid email'));
+    }
+  }
+}
+
+/**
+ * Submit handler for the order form.
+ */
+function ting_das_order_form_submit($form, &$form_state) {
+
+}
+
+/**
+ * Ajax callback for the order copy form.
+ */
+function ting_das_order_form_ajax_callback($form, &$form_state) {
+  $response = array(
+    '#type' => 'ajax',
+    '#commands' => array(),
+  );
+
+  if (is_null(form_get_errors())) {
+    $values = $form_state['values'];
+    $client = _ting_das_get_client();
+
+    $client->setUserName($values['name']);
+    $client->setUserMail($values['mail']);
+    $client->setUserLoanerId($values['loaner_id']);
+    $client->setUserInterestDate($values['interest_period']);
+    $client->setPickupAgencyId($values['fbs_preferred_branch']);
+    $client->setAgencyId(variable_get('ting_agency', ''));
+
+    $client->setPid($values['entity']['pid']);
+    $status = $client->send();
+
+    if ($status) {
+      $html = theme('status_messages') . t('Your order have been placed.');
+      $response['#commands'][] = ajax_command_ding_popup('ding_reservation', t('Success'), $html, array('refresh' => FALSE));
+    }
+    else {
+      $html = theme('status_messages') . t('Your order failed. Please try again.');
+      $response['#commands'][] = ajax_command_ding_popup('ding_reservation', t('Error'), $html, array('refresh' => FALSE));
+    }
+  }
+  else {
+    unset($form['user_login_container']['close']);
+    $html = theme('status_messages') . drupal_render($form);
+    $response['#commands'][] = ajax_command_ding_popup('ting_das', t('Order digital copy'), $html, array('refresh' => TRUE));
+  }
+
+  return $response;
+}
+
+/**
+ * Get DAS client.
+ *
+ * @return \TingDasCopyRequest
+ *   Client to communicate with the service.
+ */
+function _ting_das_get_client() {
+  static $client;
+
+  if (!isset($client)) {
+    module_load_include('inc', 'ting_das', 'includes/ting_das.clint');
+    $config = variable_get('ting_das', FALSE);
+    if (is_array($config)) {
+      $client = new TingDasCopyRequest($config['url'], $config['username'], $config['password']);
+    }
+    else {
+      watchdog('ting_das', 'The ting_das module is missing its configuration', NULL, WATCHDOG_ERROR);
```

Comment:
This has the risk of returning `NULL` and not a `TingDasCopyRequest` object.

Reply:
Added that to documentation and added code to handle it.